src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -4006,3 +4065,533 @@ function ToastSuppressor(props: any) {
         </>
     );
 }
+
+export function getDefaultClinicalDataBinFilter(
+    attribute: ClinicalAttribute
+): ClinicalDataBinFilter & {
+    showNA?: boolean | undefined;
+} {
+    return {
+        attributeId: attribute.clinicalAttributeId,
+        disableLogScale: false,
+        showNA: true,
+    } as ClinicalDataBinFilter & { showNA?: boolean };
+}
+
+export function generateColorMapKey(id: string, value: string): string {
+    return `${id}.${value}`;
+}
+
+export async function invokeGenericAssayDataCount(
+    chartInfo: GenericAssayChart,
+    filters: StudyViewFilter
+) {
+    let result: GenericAssayDataCountItem[] = [];
+
+    result = await internalClient.fetchGenericAssayDataCountsUsingPOST({
+        genericAssayDataCountFilter: {
+            genericAssayDataFilters: [
+                {
+                    stableId: chartInfo.genericAssayEntityId,
+                    profileType: chartInfo.profileType,
+                } as GenericAssayDataFilter,
+            ],
+            studyViewFilter: filters,
+        } as GenericAssayDataCountFilter,
+    });
+
+    let data = result.find(d => d.stableId === chartInfo.genericAssayEntityId);
+    let counts: ClinicalDataCount[] = [];
+    let stableId: string = '';
+    if (data !== undefined) {
+        counts = data.counts.map(c => {
+            return {
+                count: c.count,
+                value: c.value,
+            } as ClinicalDataCount;
+        });
+        stableId = data.stableId;
+    }
+
+    return { stableId: stableId, counts: counts };
```

Comment:
slightly weird that it will return empty string if no matching data is found.

Reply:
I've made it to return undefined if no data matched